api/src/main/java/org/openmrs/ui/framework/DefaultFailedAuthenticationHandler.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *  
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.ui.framework;
+
+import java.util.List;
+
+import org.openmrs.ui.framework.page.PageRequest;
+
+public class DefaultFailedAuthenticationHandler implements FailedAuthenticationHandler {
+	
+	public String handle(PageRequest pageRequest, List<String> requiredPrivileges) {
+		return getRedirectUrl();
+	}
+	
+	public String getRedirectUrl() {
+		//This is the only login page we can guarantee exists for now,
```

Comment:
This is the primary reason i didn't want to add default handler that supports context.requirePrivilege, the login page in core will soon go away after we move things to  the legacy UI module anyways